site/content/learn/playwright/codegen.md

```diff
@@ -0,0 +1,132 @@
+
+Did you know that you can use Playwright to generate your automation scripts automatically? Let's dive into how you can leverage Playwright's `codegen` feature to record scripts effortlessly.
+<!-- more -->
+
+
+### Step 1: Launch Playwright Codegen
+
+To get started, you'll use the `codegen` command from the Playwright executable. Let's say you want to create a script for interacting with [Checkly](https://checklyhq.com/). You would run the following command in your terminal:
+
+```bash
+npx playwright codegen https://checklyhq.com
+```
+
+When you execute this command, two windows will open up. On the left, you have a browser window displaying the target site, and on the right, you see the Playwright inspector.
+
+![Playwright codegen inspector](/learn/images/codegen-1.png)
+
+Note that the inspector will populate with the basic code needed for a Playwright test, and start with the line needed to load the url you gave in the terminal.
+
+Also, the browser will automatically highlight elements, and show you a locator for them in Playwright’s preferred [user-first selector](https://www.checklyhq.com/blog/playwright-user-first-selectors/) format. 
+
+### Step 2: Interact with the Website
+
+Now, you can start interacting with the website as you normally would. For example:
+
+1. Accept cookies.
+2. Click a few links.
+3. Type "hello world" in a search box.
+
+As you perform these actions, you'll notice that the Playwright inspector on the right is recording every interaction in real-time. Each action is converted into a line of code, creating a script as you go. For example the steps above would produce the code as follows:
+
+```ts
+test('test', async ({ page }) => {
+  await page.goto('https://www.checklyhq.com/');
+  await page.getByRole('button', { name: 'Accept All' }).click();
+  await page.getByRole('button', { name: 'Developers' }).first().click();
+  await page.getByRole('link', { name: 'Documentation Technical docs' }).click();
+  await page.getByLabel('Search').click();
+  await page.getByPlaceholder('Search docs').click();
+  await page.getByPlaceholder('Search docs').fill('hello world');
+});
+```
+
+Note that codegen will make a ‘best effort’ at emulating your actions and writing ‘best practice’ code, so you should always review it before deploying.
+
+### Step 5: Add Assertions to your tests
+
+Every action in the script above has implicit assertions about the elements we interact with. Playwright will expect these elements are stable, visible, and enabled and if one isn’t then the test will will fail we could call this done, but we generally want our tests to contain assertions beyond just navigating. There are a few reasons why you should have assertions in your test: It makes it clearer to future coders what your test was checking for, and the error message is more useful from a failed assertion rather than just failing to find an element. Right now all we’ll get if the test above fails is a timeout and a line which failed. 
```

Comment:
The 2nd sentence seems very long and includes a `will will`. :)

Reply:
revised